pkg/services/ngalert/store/loki_range_to_instant.go

```diff
@@ -0,0 +1,60 @@
+package store
+
+import (
+	"encoding/json"
+
+	"github.com/grafana/grafana/pkg/services/ngalert/models"
+)
+
+const (
+	grafanaCloudLogs          = "grafanacloud-logs"
+	grafanaCloudUsageInsights = "grafanacloud-usage-insights"
+)
+
+func canBeInstant(r *models.AlertRule) bool {
+	if len(r.Data) < 2 {
+		return false
+	}
+	// First query part should be range query.
+	if r.Data[0].QueryType != "range" {
+		return false
+	}
+	// First query part should go to cloud logs or insights.
+	if r.Data[0].DatasourceUID != grafanaCloudLogs && r.Data[0].DatasourceUID != grafanaCloudUsageInsights {
+		return false
+	}
+	// Second query part should be and expression
+	if r.Data[1].DatasourceUID != "__expr__" {
```

Comment:
These are sometimes encoded as `-100`, it's a legacy thing but certainly still around in some places

Reply:
Ah, good shout.